src/Integration/ModulesIntegration.php

```diff
@@ -21,13 +22,28 @@ final class ModulesIntegration implements IntegrationInterface
      */
     private static $loadedModules = [];
 
+    /**
+     * @var HubInterface The hub this integration is attached to
+     */
+    private $hub;
+
+    /**
+     * Constructor.
+     *
+     * @param HubInterface|null $hub The hub this integration is attached to
+     */
+    public function __construct(?HubInterface $hub = null)
+    {
+        $this->hub = $hub ?? HubAdapter::getInstance();
+    }
+
     /**
      * {@inheritdoc}
      */
     public function setupOnce(): void
     {
         Scope::addGlobalEventProcessor(function (Event $event) {
-            $self = Hub::getCurrent()->getIntegration(self::class);
+            $self = $this->hub->getIntegration(self::class);
```

Comment:
Checking if the integration is active on the current client, we need to ask for the current hub on the time when this is called. It's not valid to pass it into the Ctor and rely on this instance since this Hub can already be gone.